apps/src/componentLibrary/checkboxDropdown/CheckboxDropdown.tsx

```diff
@@ -0,0 +1,193 @@
+import React, {
+  useCallback,
+  useMemo,
+  useRef,
+  useEffect,
+  KeyboardEvent,
+} from 'react';
+import classNames from 'classnames';
+
+import {ComponentSizeXSToL} from '@cdo/apps/componentLibrary/common/types';
+import moduleStyles from './checkboxDropdown.module.scss';
+import Button from '@cdo/apps/templates/Button';
+
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import i18n from '@cdo/locale';
+
+import Checkbox from '@cdo/apps/componentLibrary/checkbox';
+import FontAwesomeV6Icon from '@cdo/apps/componentLibrary/fontAwesomeV6Icon';
+import {
+  DropdownProvider,
+  useDropdownContext,
+} from '@cdo/apps/componentLibrary/common/contexts/DropdownContext';
+
+export interface CheckboxDropdownProps {
+  /** CheckboxDropdown name */
+  name: string;
+  /** CheckboxDropdown color */
+  color?: 'white' | 'black';
+  /** CheckboxDropdown size */
+  size: ComponentSizeXSToL;
+  /** CheckboxDropdown disabled state */
+  disabled?: boolean;
+  /** CheckboxDropdown label */
+  labelText: string;
+  /** CheckboxDropdown options */
+  allOptions: {value: string; label: string; isOptionDisabled?: boolean}[];
+  /** CheckboxDropdown checked options */
+  checkedOptions: string[];
+  /** CheckboxDropdown onChange handler */
+  onChange: (args: React.ChangeEvent<HTMLInputElement>) => void;
+  /** CheckboxDropdown onSelectAll handler */
+  onSelectAll: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  /** CheckboxDropdown onClearAll handler */
+  onClearAll: (event: React.MouseEvent<HTMLButtonElement>) => void;
+}
+
+/**
+ * ### Production-ready Checklist:
+ * * (✔) implementation of component approved by design team;
+ * * (✔) has storybook, covered with stories and documentation;
+ * * (✔) has tests: test every prop, every state and every interaction that's js related;
+ * * (see apps/test/unit/componentLibrary/DropdownMenuTest.jsx)
+ * * (?) passes accessibility checks;
+ *
+ * ###  Status: ```Ready for dev```
+ *
+ * Design System: Checkbox Dropdown Component.
+ * Used to render checkbox (multiple choice) dropdowns.
+ */
+const CheckboxDropdown: React.FunctionComponent<CheckboxDropdownProps> = ({
+  name,
+  labelText,
+  allOptions,
+  checkedOptions = [],
+  onChange,
+  onSelectAll,
+  onClearAll,
```

Comment:
Why do we pass this in as a prop (vs defining it in this component)? Isn't it kind of expected that a clear all button will...clear all? Same question re: onSelectAll

Reply:
Those are added to make component fully controllable from the parent component

Reply:
Yeah Ben at first I had the same thought, but I do agree that deferring control to the parent would probably be better overall. I'm thinking of how this would probably be used with redux pretty often - in which case the clear all function would just look something like `const onClearAll = () => dispatch(setSelectedOptions([]));`

Reply:
Ok, fair enough. I guess my concern is that I think my preference for these super reusable components is that they emphasize ease of use rather than customization, whereas I think this approach emphasizees the latter.